Pcap++/src/PcapFilter.cpp

```diff
@@ -297,48 +297,43 @@ void EtherTypeFilter::parseToString(std::string& result)
 	result = "ether proto " + stream.str();
 }
 
-AndFilter::AndFilter(std::vector<GeneralFilter*>& filters)
+CompositeFilter::CompositeFilter(std::vector<GeneralFilter*>& filters, std::string compositeDelimiter)
+	: m_CompositeDelimiter(std::move(compositeDelimiter))
 {
 	for(std::vector<GeneralFilter*>::iterator it = filters.begin(); it != filters.end(); ++it)
 	{
 		m_FilterList.push_back(*it);
 	}
 }
 
-void AndFilter::setFilters(std::vector<GeneralFilter*>& filters)
+void CompositeFilter::removeFilter(GeneralFilter* filter)
 {
-	m_FilterList.clear();
-
-	for(std::vector<GeneralFilter*>::iterator it = filters.begin(); it != filters.end(); ++it)
+	std::vector<GeneralFilter*>::iterator it = m_FilterList.begin();
```

Comment:
this part can be shortened to:

```cpp
	for (auto it = m_FilterList.begin(); it != m_FilterList.end(); ++it)
	{
		if (*it == filter)
		{
			m_FilterList.erase(it);
			break;
		}
	}
```

Reply:
Resolved in 7f6dba6 and dde74b9
Iterator type is left as `std::vector<GeneralFilter*>::iterator` instead of `auto` for consistency with the other functions.